File: server/models/Interview.js
const mongoose = require("mongoose");

const interviewSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
  resume: { type: mongoose.Schema.Types.ObjectId, ref: "Resume" },
  job: { type: mongoose.Schema.Types.ObjectId, ref: "Job" },
  difficulty: String,
  mode: String,
  status: { type: String, default: "in-progress" }
}, { timestamps: true });

module.exports = mongoose.model("Interview", interviewSchema);

File: server/models/interview.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from bson import ObjectId

class InterviewBase(BaseModel):
    user: str
    resume: Optional[str] = None
    job: Optional[str] = None
    difficulty: str
    mode: str
    status: Optional[str] = "in-progress"

class InterviewCreate(InterviewBase):
    pass

class Interview(InterviewBase):
    id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
        json_encoders = {ObjectId: str}